### PR TITLE
Updates for probe-rs master

### PR DESCRIPTION
**Pull request by MabezDev**
* Update generation code to use the new master format.
* Fix up update to use the new `load_address`, and offset the date
section.

### CODE REVIEW

**Comment by Tiwalun** [comment]
Please rebase onto master, CI should be working again then.

File: src/generate.rs [hunk 1]
```diff
@@ -367,6 +377,7 @@ pub(crate) fn get_ram(device: &Device) -> Option<RamRegion> {
             regions.push(RamRegion {
                 range: memory.start as u32..memory.start as u32 + memory.size as u32,
                 is_boot_memory: memory.startup,
+                cores: vec!["".to_owned()], // TODO can we do better?
```
Comment (Yatekii) [inline src/generate.rs:380]: Yes we actually need to do better for it to work. It needs to know which core it should use to access this memory. As a default we can use "main". See here: https://github.com/probe-rs/probe-rs/blob/master/probe-rs/targets/nRF52_Series.yaml#L21. The problem is that you use the actual core name, which is not main, so I think we need to change one or the other :) Currently you use the core type as the name which is not a good choice as the multi targets can have multiple cores, even tho you do not implement multi core handling yet.

**Review by Yatekii — APPROVED** [review]
I think this can be merged if you do not want to fix the remaining things. I planned to do that anyways, but you know the empty promises :/
We surely have to fix all the multicore handling but this will generate a better target file which then can be tweaked to run well :)

bors d=MabezDev

File: src/generate.rs (hunk 1 above)
Comment (MabezDev) [reply src/generate.rs]: Switched to using "main" as the default core for everything, don't have time to look at multicore atm, sorry!

**Comment by MabezDev** [comment]
Oof

File: src/generate.rs (hunk 1 above)
Comment (Yatekii) [reply src/generate.rs]: No need to be sorry! I am very thankful you did this already!!

**Comment by Yatekii** [comment]
Great now it locked up lol :D

**Review by Tiwalun — APPROVED** [review]

**Comment by Tiwalun** [comment]
> 
> 
> Great now it locked up lol :D

Just the review was missing, then you can merge it normally.

**Comment by Yatekii** [comment]
Ah lol, it said bors failed and I thought I can't really change that. But I just had a quick look.

**Pull request merged**